lib/pbkit/pbkit.c

```diff
@@ -2290,160 +2291,97 @@ void pb_end(DWORD *pEnd)
 }
 
 
-
-void pb_push1to(DWORD subchannel, DWORD *p, DWORD command, DWORD param1)
+inline DWORD *pb_push_to(DWORD subchannel, DWORD *p, DWORD command, DWORD nparam)
 {
 #ifdef DBG
-    if (p!=pb_PushNext) debugPrint("pb_push1to: new write address invalid or not following previous write addresses\n");
-    if (pb_BeginEndPair==0) debugPrint("pb_push1to: missing pb_begin earlier\n");
-    pb_PushIndex+=2;
-    pb_PushNext+=2;
-    if (pb_PushIndex>128) debugPrint("pb_push1to: begin-end block musn't exceed 128 dwords\n");
+    if (p!=pb_PushNext) debugPrint("pb_push_to: new write address invalid or not following previous write addresses\n");
+    if (pb_BeginEndPair==0) debugPrint("pb_push_to: missing pb_begin earlier\n");
+    pb_PushIndex += 1 + nparam;
+    pb_PushNext += 1 + nparam;
+    if (pb_PushIndex>128) debugPrint("pb_push_to: begin-end block musn't exceed 128 dwords\n");
 #endif
 
-    *(p+0)=EncodeMethod(subchannel,command,1);
-    *(p+1)=param1;
+    *(p+0)=EncodeMethod(subchannel,command,nparam);
+    return p+1+nparam;
 }
 
-void pb_push2to(DWORD subchannel, DWORD *p, DWORD command, DWORD param1, DWORD param2)
+inline DWORD *pb_push1_to(DWORD subchannel, DWORD *p, DWORD command, DWORD param1)
 {
-#ifdef DBG
-    if (p!=pb_PushNext) debugPrint("pb_push2to : new write address invalid or not following previous write addresses\n");
-    if (pb_BeginEndPair==0) debugPrint("pb_push2to : missing pb_begin earlier\n");
-    pb_PushIndex+=3;
-    pb_PushNext+=3;
-    if (pb_PushIndex>128) debugPrint("pb_push2to: begin-end block musn't exceed 128 dwords\n");
-#endif
+    pb_push_to(subchannel,p,command,1);
+    *(p+1)=param1;
+    return p+2;
+}
 
-    *(p+0)=EncodeMethod(subchannel,command,2);
+inline DWORD *pb_push2_to(DWORD subchannel, DWORD *p, DWORD command, DWORD param1, DWORD param2)
+{
+    pb_push_to(subchannel,p,command,2);
```

Comment:
Return value ignored

Reply:
This is intended: `pb_push_to` returns `p+3` - it includes the gap for the data (which is written by this function).

So `pb_push_to` writes `*(p+0)` and this function writes the rest using the old pointer; on return, this function does the same increment that `pb_push_to` did (which was discarded).

I agree that this code is confusing.

I guess an alternative is:

```c
uint32_t* p_new = pb_push_to(subchannel,p,command,2);
*(p+1) = param1;
*(p+2) = param2;
return p_new;
```

*(I don't think that helps very much though)*

Reply:
Taken care of now - `pb_push` and `pb_push_to` now return `void`.